src/JwksClient.js

```diff
@@ -78,20 +78,25 @@ export class JwksClient {
             }  
             return ((key.x5c && key.x5c.length) || (key.n && key.e));
         })
-        .map(key => {
-          if (key.x5c && key.x5c.length) {
-            return {
-              kid: key.kid,
-              nbf: key.nbf,
-              publicKey: certToPEM(key.x5c[0])
-            };
-          } else {
-            return {
-              kid: key.kid,
-              nbf: key.nbf,
-              rsaPublicKey: rsaPublicKeyToPEM(key.n, key.e)
-            };
-          }
+          .map(key => {
+            const hasCertificateChain = key.x5c && key.x5c.length;
+            if (hasCertificateChain){
+              const pem = certToPEM(key.x5c[0]);
+              return {
+                kid: key.kid,
+                nbf: key.nbf,
+                publicKey: pem,
+                getPublicKey: () => pem
```

Comment:
Might be safer to return this.publicKey.  That way if something else changes the `publicKey` property getPublicKey isn't off doing it's own thig.  Basically prevents two separate instances in -memory.  Right now this is creating a closure.

Reply:
The approach we agreed on Friday doesn't work because the `publicKey` property is not yet defined.

```js
return {
  kid: key.kid,
  nbf: key.nbf,
  publicKey: pem,
  getPublicKey: () => this.publicKey //returns undefined
};
```

The alternative was to first create the base object with common props, and later depending on the IF branch, add the additional prop with the same value. That is what I did on the first commit (with the difference of using a prop and not a function) that you asked me to refactor. Should I roll back to that first commit or keep this as is now?

Reply:
What about 
```js
const jwk = {};
if (hasCertificateChain){
  jwk.kid = key.kid;
  jwk.nbf = key.nbf;
  jwk.publicKey = certToPEM(key.x5c[0]);
  jwk.getPublicKey = () => jwk.publicKey;
} else {
  jwk.kid = key.kid;
  jwk.nbf = key.nbf;
  jwk.rsaPublicKey = rsaPublicKeyToPEM(key.n, key.e);
  jwk.getPublicKey = () => jwk.rsaPublicKey;
}
return jwk;
```

Reply:
I just did, it doesn't work. It's still saying the result from calling that function is `undefined`. 

We can pair tomorrow there's no rush 👍

Reply:
You're right, let's go with your most recent suggestion.  At that point you may as well move the kid/nbf setting to the section above as per your earlier PR.